.github/scripts/sign.ps1

```diff
@@ -0,0 +1,135 @@
+param(
+    [Parameter(Mandatory=$true)] [string]$clientId,
+    [Parameter(Mandatory=$true)] [string]$workspace,
+    [Parameter(Mandatory=$true)] [string]$subscriptionId,
+    [Parameter(Mandatory=$true)] [string]$storage_name,
+    [Parameter(Mandatory=$true)] [string]$container_name,
+    [Parameter(Mandatory=$true)] [string]$vault_name,
+    [Parameter(Mandatory=$true)] [string]$aad_cert,
+    [Parameter(Mandatory=$true)] [string]$sign_cert,
+    [Parameter(Mandatory=$true)] [string]$signing_cert_fingerprint,
+    [Parameter(Mandatory=$false)] [switch]$user_login
+)
+
+$fileName = (Get-ChildItem -Recurse -Filter *.nupkg | Select-Object -Property Name -First 1).Name
+if ($fileName -match ' ') {
+    # This accounts for cases where it finds the same package and we end up with $fileName = "proj.1.1.0.nupkg proj.1.1.0.nupkg"
+    $fileName = $fileName.split()[0]
+}
+
+if ([string]::IsNullOrWhiteSpace($fileName)) {
+    throw "Unable to find nupkg for signing. Ensure the 'dotnet pack' command has been run and that it's output to a directory called 'unsigned'."
+}
+
+Write-Host "Found nupkg: $fileName"
+if ($user_login) {
```

Comment:
Shouldn't we -instead- check if the value is missing and throw exception?

Reply:
It's a little confusing what you're referring to with the lines you highlighted so I'll try and answer to both..
1. If you're referring to the nupkg stuff - that is exactly what it's currently doing. If it can't find a nupkg it throws an exception.
2. If you're referring to the `if ($user_login)` - no, I only have this in there for local testing. The logging in/out stuff is run in the pipeline under normal circumstances.